### PR TITLE
Add Table metadata cache expiration configuration

### PR DESCRIPTION
**Pull request by brfrn169**
This PR allows us to set Table metadata cache expiration configuration with the `scalar.db.table_metadata.cache_expiration_time_secs` property. Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/storage/common/TableMetadataManager.java [hunk 1]
```diff
@@ -18,16 +19,19 @@
   private final LoadingCache<TableKey, Optional<TableMetadata>> tableMetadataCache;
 
   public TableMetadataManager(DistributedStorageAdmin admin, DatabaseConfig config) {
+    CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder();
+    long tableMetadataCacheExpirationTimeSecs = config.getTableMetadataCacheExpirationTimeSecs();
+    if (tableMetadataCacheExpirationTimeSecs > 0) {
+      builder.expireAfterWrite(tableMetadataCacheExpirationTimeSecs, TimeUnit.SECONDS);
```
Comment (feeblefakie) [inline core/src/main/java/com/scalar/db/storage/common/TableMetadataManager.java:25]: I'm not really sure about the use case of specifying the expiration time. Can you clarify?
From users' perspective, it seems like it should be cached forever or it should be replaced right after it needs to 
 be.

**Review by feeblefakie — COMMENTED** [review]
Thank you!
I left one comment since I'm not sure about the use case.

File: core/src/main/java/com/scalar/db/storage/common/TableMetadataManager.java (hunk 1 above)
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/storage/common/TableMetadataManager.java:25]: The use case is when the table metadata change happens. As we don't have `alter table` feature for now, `create table` and `delete table` are only the use cases. Anyway, I think we will need to handle `alter table` cases in the future, so this cache expiration feature will become more important.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
But the CIs are failing. Can you take a look?

**Comment by brfrn169** [comment]
They were successful after rerunning the CIs. I'm merging this.

**Pull request merged**